[TasmiaH0508] iP

TasmiaH0508 opened the pull request:
### Bob the ChatBot

> "Your mind is for having ideas, not holding them." – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Bob the ChatBot frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

Features:

- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

TasmiaH0508 committed:
Rename chatbot to Bob. Add exit function for chatbot.

TasmiaH0508 committed:
Add echo function to chatbot

TasmiaH0508 committed:
Add addTask and display taskList functions.

TasmiaH0508 committed:
Add functions to mark tasks as done or not done

TasmiaH0508 committed:
Add tracking function for event, todo and deadline

TasmiaH0508 committed:
Perform automated text UI testing

TasmiaH0508 committed:
Tidy code to make code more readable

TasmiaH0508 committed:
Handle incorrect inputs error

TasmiaH0508 committed:
Add delete task functionality

TasmiaH0508 committed:
Implement part of Level-7

TasmiaH0508 committed:
Complete level-7

TasmiaH0508 committed:
Merge branch-Level-7 into master

TasmiaH0508 committed:
Complete Level 8

TasmiaH0508 committed:
Merge branch 'branch-Level-8'
Merge branch-Level-7 into master

TasmiaH0508 committed:
Add Storage class

TasmiaH0508 committed:
Package classes

TasmiaH0508 committed:
Merge branch 'add-gradle-support'
Add gradle support

TasmiaH0508 committed:
Add JUnit tests

TasmiaH0508 committed:
Tidy code to follow coding standard

TasmiaH0508 committed:
Add java doc

TasmiaH0508 committed:
Add search feature

TasmiaH0508 committed:
Update .gitignore file

TasmiaH0508 committed:
Resolve merge conflicts from branch-A-CodingStandard

TasmiaH0508 committed:
Resolve merge conflicts between master and branch-Level-9

TasmiaH0508 committed:
Resolve merge conflicts between master and branch-Level-9

TasmiaH0508 committed:
Re-implement search

TasmiaH0508 committed:
Add java doc

TasmiaH0508 committed:
Add javadoc

TasmiaH0508 committed:
Complete Level 10

TasmiaH0508 committed:
Merge branch-Level-10 into master

SheYuting commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,56 @@
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+
```
Do remember to provide a JavaDoc description for a public class

SheYuting commented on src/main/java/bob/tasklist/TaskList.java:
```diff
@@ -99,65 +102,80 @@ public void createAndAddTask(Parser p, String s) throws IndexOutOfBoundsExceptio
      * @param index Index position of task in list.
      */
-    public void deleteTask(int index, Parser p) {
+    public String deleteTask(int index, Parser p) {
```
Maybe do not name the input parameters as a single alphabet is a better idea here!

SheYuting reviewed (commented):
Overall, I like how you add JavaDoc Comments on most of the public methods! Good job!

jaylontan commented on src/main/java/bob/tasks/Task.java:
```diff
@@ -0,0 +1,68 @@
+     *
+     * @return "[X]" if task is complete and "[ ]" otherwise.
+     */
+    public String getStatusIcon() {
+        if (isComplete) {
+            return "[X]";
+        } else {
+            return "[ ]";
+        }
+    }
+
+    /**
+     * Returns task description.
+     *
+     * @return taskname prefixed by status icon.
+     */
+    public String getTaskDescription() {
+        return getStatusIcon() + this.taskName;
+    }
+
+    /**
+     * Returns task name without status icon.
+     *
+     * @return taskname.
+     */
+    public String getTaskDescriptionWOIcon() {
+        return this.taskName;
+    }
```
Good function to separate marked icon with description in the case where only the task name is needed.

jaylontan commented on src/main/java/bob/parser/Parser.java:
```diff
@@ -0,0 +1,111 @@
+     * @param s S string containing substrings separated by spaces.
+     * @return array of substrings
+     */
+    public String[] splitStringBySpacing(String s) {
+        return s.split(" ");
+    }
```
Better to use more descriptive parameter names rather than a single letter

jaylontan commented on src/main/java/bob/Bob.java:
```diff
@@ -0,0 +1,335 @@
+     * @param t T command type.
+     * @return string depending on input t
+     */
+    public static String enumCommandTypeToString(CommandType t) {
+        if (t == CommandType.MARK) {
+            return "mark";
+        } else if (t == CommandType.TODO) {
+            return "todo";
+        } else if (t == CommandType.DEADLINE) {
+            return "deadline";
+        } else if (t == CommandType.EVENT) {
+            return "event";
+        } else if (t == CommandType.BYE) {
+            return "bye";
+        } else if (t == CommandType.LIST) {
+            return "list";
+        } else if (t == CommandType.DELETE) {
+            return "delete";
+        } else {
+            return "search";
+        }
+    }
```
Consider using switch case statement instead of multiple if-else blocks

jaylontan commented on src/main/java/bob/Bob.java:
```diff
@@ -0,0 +1,335 @@
+    private static Parser parser = new Parser();
+    private static boolean isEndConversation = false;
+    private static final TaskList TASK_LIST = new TaskList();
+    private static final String[] COMMANDS = {"mark", "todo", "deadline", "event" , "bye", "list", "delete", ""};
```
Uses constants instead of hardcoded values, which makes it easier to modify and maintain. Good!

TasmiaH0508 committed:
Squash bugs related to search

TasmiaH0508 committed:
Remove unneccessary file and check for code quality issues

TasmiaH0508 committed:
Apply assertions and check code standards

Most of code standards were checked and violations were fixed while in branch-A-Assertions to improve code quality.

Assertions were applied where neccessary (mainly in the Bob.java file). Doing so allows for improved safety.

TasmiaH0508 committed:
Squash bug

When user keyed in an incorrect date format, program exited due to runtime exception.
Bug squashed so that such runtime exception will not happen again.

TasmiaH0508 committed:
Merge pull request #1 from TasmiaH0508/branch-A-CodeQuality

Remove unneccessary file and check for code quality issues

TasmiaH0508 committed:
Merge branch 'master' into branch-A-Assertions
Merge master branch onto branch-A-Assertions

TasmiaH0508 committed:
Merge pull request #2 from TasmiaH0508/branch-A-Assertions

Branch a assertions

TasmiaH0508 committed:
Tidy code

TasmiaH0508 committed:
Implement mass-delete and mass-marking of tasks

TasmiaH0508 committed:
Merge branch 'branch-BCD-extension'
Merge branch-BCD-extension to master

TasmiaH0508 committed:
Add clean command to remove completed tasks

TasmiaH0508 committed:
Fix package name to follow convention

TasmiaH0508 committed:
Fix import statements

TasmiaH0508 committed:
Rename chatbot

TasmiaH0508 committed:
Update project README

TasmiaH0508 committed:
Debug search method

TasmiaH0508 committed:
Update docs/README

TasmiaH0508 committed:
Add Ui.png

TasmiaH0508 committed:
Update docs/README.md

TasmiaH0508 committed:
Rename file

TasmiaH0508 committed:
Rename file

TasmiaH0508 committed:
Squash bugs related to invalid input

TasmiaH0508 committed:
Update docs/README.md

TasmiaH0508 committed:
Make window size adjustable

TasmiaH0508 committed:
Add color

TasmiaH0508 committed:
Add background image

TasmiaH0508 committed:
Optimise code

TasmiaH0508 committed:
Make error messages more helpful

TasmiaH0508 committed:
Update user guide with new images
